testsuite/integration/elytron-oidc-client/src/test/java/org/wildfly/test/integration/elytron/oidc/client/logout/OidcLogoutBaseTest.java

```diff
@@ -0,0 +1,521 @@
+                TestSuiteEnvironment.getHttpPort(),
+                "/" + appName + SimpleSecuredServlet.SERVLET_PATH
+                        + config.getLogoutPath()).toURI();
+        logoutOfKeycloak(requestUri, HttpURLConnection.HTTP_OK, expectedText, true);
+    }
+
+    public static void logoutOfKeycloak(URI requestUri, int expectedStatusCode, String expectedText,
+                                        boolean logoutFromKeycloak) throws Exception {
+
+        HttpContext context = new BasicHttpContext();
+        HttpResponse response = null;
+        HttpGet getMethod = new HttpGet(requestUri);
+        response = httpClient.execute(getMethod, context);
+
+        try {
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (logoutFromKeycloak) {
+                assertTrue("Expected code == OK but got " + statusCode + " for request=" + requestUri, statusCode == HttpURLConnection.HTTP_OK);
+                response.getEntity();
+                String responseString = new BasicResponseHandler().handleResponse(response);
+                assertTrue("Unexpected result " + expectedText, responseString.contains(expectedText));
+            }
+            else {
+                assertTrue("Expected code == FORBIDDEN but got " + statusCode + " for request=" + requestUri, statusCode == HttpURLConnection.HTTP_FORBIDDEN);
+            }
+        } finally {
+            HttpClientUtils.closeQuietly(response);
+        }
+    }
+
+    public static void assertUserLoggedIn(String appName, String expectedText) throws Exception {
+        accessPage(generateURL(appName).toURI(), HttpURLConnection.HTTP_OK, expectedText);
+    }
+
+    public static void assertUserLoggedOut(String appName, String expectedText) throws Exception {
+        accessPage(generateURL(appName).toURI(), HttpURLConnection.HTTP_OK, expectedText);
+    }
+
+    public static void accessPage(URI requestUri, int expectedStatusCode,
+                                     String expectedText) throws Exception {
+        HttpContext context = new BasicHttpContext();
+        HttpResponse response = null;
+        HttpGet getMethod = new HttpGet(requestUri);
+        response = httpClient.execute(getMethod, context);
+
+        try {
+            int statusCode = response.getStatusLine().getStatusCode();
+            assertTrue("Expected code == " + expectedStatusCode + " but got "
+                            + statusCode + " for request=" + requestUri,
+                    statusCode == expectedStatusCode);
+            response.getEntity();
+            String responseString = new BasicResponseHandler().handleResponse(response);
+            assertTrue("Unexpected result " + expectedText,
+                    responseString.contains(expectedText));
+        } finally {
+            HttpClientUtils.closeQuietly(response);
+        }
+    }
+
+    public static class KeycloakSetup implements ServerSetupTask {
```

Comment:
This class and the code below is also a duplicate of code in OidcBaseTest, right? Would it make sense to have common code that both classes could make use of?